tests/atomic/main.yml

```diff
@@ -0,0 +1,527 @@
+---
+# vim: set ft=ansible:
+#
+- name: Atomic - Setup
+  hosts: all
+  become: yes
+
+  tags:
+    - setup
+
+  vars_files:
+    - vars.yml
+
+  roles:
+    # This playbook requires Ansible 2.2 and an Atomic Host
+    - role: ansible_version_check
+      avc_major: "2"
+      avc_minor: "2"
+      tags:
+        - ansible_version_check
+
+    # Subscribe if the system is RHEL
+    - role: redhat_subscription
+      when: ansible_distribution == 'RedHat'
+      tags:
+        - redhat_subscription
+
+- name: Atomic Fully Qualified Name
+  hosts: all
+  become: yes
+
+  tags:
+    - fq_name
+
+  vars_files:
+    - vars.yml
+
+  pre_tasks:
+    - name: Set cockpit container name for RHELAH
+      set_fact:
+        cockpit_cname: "registry.access.redhat.com/rhel7/cockpit-ws"
+      when: ansible_distribution == "RedHat"
+
+    - name: Set cockpit container name for Fedora/CentOS
+      set_fact:
+        cockpit_cname: "registry.fedoraproject.org/f25/cockpit"
+      when: "'CentOS' in ansible_distribution or ansible_distribution == 'Fedora'"
+
+  roles:
+    - role: atomic_pull
+      apl_image: "{{ cockpit_cname }}"
+
+    - role: atomic_images_list_verify
+      expected_values:
+        repo: "{{ cockpit_cname }}"
+
+    - role: atomic_containers_list_verify
+      expected_values:
+        image_name: "{{ cockpit_cname }}"
+      expect_missing: true
+
+    - role: atomic_install
+      ai_image: "{{ cockpit_cname }}"
+
+    - role: atomic_run
+      ar_image: "{{ cockpit_cname }}"
+
+    - role: atomic_images_list_verify
+      expected_values:
+        repo: "{{ cockpit_cname }}"
+
+    - role: atomic_containers_list_verify
+      expected_values:
+        image_name: "{{ cockpit_cname }}"
+      check_mode: false
+
+    - role: atomic_stop
+      as_container: "{{ aclv_acl_jq_match['id'] }}"
+
+    - role: atomic_containers_delete
+      acd_container: "{{ aclv_acl_jq_match['id'] }}"
+
+    - role: atomic_images_list_verify
+      expected_values:
+        repo: "{{ cockpit_cname }}"
+
+    - role: atomic_containers_list_verify
+      expected_values:
+        image_name: "{{ cockpit_cname }}"
+      expect_missing: true
+
+    - role: atomic_uninstall
+      au_image: "{{ cockpit_cname }}"
+
+    - role: atomic_images_delete
+      aid_image: "{{ ailv_match['id'] }}"
+
+    - role: atomic_images_list_verify
+      expected_values:
+        repo: "{{ cockpit_cname }}"
+      expect_missing: true
+
+    - role: atomic_containers_list_verify
+      expected_values:
+        image_name: "{{ cockpit_cname }}"
+      expect_missing: true
+
+- name: Atomic Short Name
+  hosts: all
+  become: yes
+
+  tags:
+    - shortname
+
+  vars_files:
+    - vars.yml
+
+  pre_tasks:
+    - name: Set cockpit container name for RHELAH
```

Comment:
Maybe move these types of distro specific facts into vars files and use `with_first_found` to load them in?

Reply:
I ended up using vars_files with the ansible_distribution name since there was no difference between the different versions of a distribution (ie fedora 26, 27, etc).  The only thing I had to account for the ansible_distribution for cahc which is centosdev.  If needed I could use include_vars with with_first_found.  I didn't feel that was needed for this simple case.